metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1701,6 +1691,37 @@ class MetalsLanguageServer(
           Future.successful(()).asJavaObject
         }
 
+      case ServerCommands.InsertInferredType() =>
+        CancelTokens.future { token =>
+          val args = params.getArguments().asScala
+          val futureOpt = for {
+            arg0 <- args.lift(0)
+            uri <- Argument.getAsString(arg0)
+            arg1 <- args.lift(1)
+            line <- Argument.getAsInt(arg1)
+            arg2 <- args.lift(2)
+            character <- Argument.getAsInt(arg2)
+            pos = new l.Position(line, character)
+            textDoc = new l.TextDocumentIdentifier(uri)
+            params = new TextDocumentPositionParams(textDoc, pos)
+          } yield {
+            for {
+              edits <- compilers.insertInferredType(params, token)
+              if (!edits.isEmpty())
+              workspaceEdit = new l.WorkspaceEdit(Map(uri -> edits).asJava)
+              _ <- languageClient
+                .applyEdit(new ApplyWorkspaceEditParams(workspaceEdit))
+                .asScala
+            } yield ()
+          }
+
+          futureOpt
+            .getOrElse {
+              languageClient.showMessage(Messages.InsertInferredTypeFailed)
+              Future.unit
+            }
+            .asInstanceOf[Future[Object]]
```

Comment:
A nitpick triggered by `asInstanceOf`
```suggestion
          val paramsOpt = for {
            arg0 <- args.lift(0)
            uri <- Argument.getAsString(arg0)
            arg1 <- args.lift(1)
            line <- Argument.getAsInt(arg1)
            arg2 <- args.lift(2)
            character <- Argument.getAsInt(arg2)
            pos = new l.Position(line, character)
            textDoc = new l.TextDocumentIdentifier(uri)
            params = new TextDocumentPositionParams(textDoc, pos)
          } yield {
            for {
              edits <- compilers.insertInferredType(params, token)
              if (!edits.isEmpty())
              workspaceEdit = new l.WorkspaceEdit(Map(uri -> edits).asJava)
              _ <- languageClient
                .applyEdit(new ApplyWorkspaceEditParams(workspaceEdit))
                .asScala
            } yield ()
          }
          
          paramsOpt.fold(Future.unit){ params =>
              for {
                 edits <- compilers.insertInferredType(params, token)
                 if (!edits.isEmpty())
                 workspaceEdit = new l.WorkspaceEdit(Map(uri -> edits).asJava)
                 _ <- languageClient
                   .applyEdit(new ApplyWorkspaceEditParams(workspaceEdit))
                   .asScala
                } yield ()
          }.asJavaObject
```

Reply:
`asJavaObject` would need `.asScala`, so wrapping it even more. And anyway asInstanceOf is hapening in `asJavaObject`

Reply:
Yes, but it the first time when it will be used directly in this file. Other actions are using `asJavaObject`. I think it would be better to preserve this kind of consistency.

Reply:
Added a method `unitToJavaObject` that just does the cast. What do you think?

Reply:
Or actually I think `withObjectValue` makes more sense.

Reply:
Looks good!